python/common/python/math/__init__.java

```diff
@@ -0,0 +1,840 @@
+            __doc__ = "",
+            args = {"arg"}
+    )
+    public static org.python.Object tanh(org.python.Object arg) {
+        double v = toFloatValue(arg);
+        if (exceptInf(Math.tanh(v), v)) {
+            throw new org.python.exceptions.OverflowError("math range error");
+        }
+        return new org.python.types.Float(Math.tanh(v));
+    }
+
+
+    @org.python.Method(
+            __doc__ = "",
+            args = {"arg"}
+    )
+    public static org.python.Object sqrt(org.python.Object arg) {
+        double v = toFloatValue(arg);
+        if (exceptNaN(Math.sqrt(v), v)) {
+            throw new org.python.exceptions.ValueError("math domain error");
+        } else {
+            return new org.python.types.Float(Math.sqrt(v));
+        }
+    }
+
+
+    @org.python.Method(
+            __doc__ = "",
+            args = {"arg1", "arg2"}
+    )
+    public static org.python.Object gcd(org.python.Object arg1, org.python.Object arg2) {
+        if (org.Python.VERSION < 0x03050000) {
+            throw new org.python.exceptions.AttributeError("'module' object has no attribute 'gcd'");
+        } else {
+            if (!(arg1 instanceof org.python.types.Int)) {
+                throw new org.python.exceptions.TypeError("'" + arg1.typeName() + "' object cannot be interpreted as an integer");
+            } else if (!(arg2 instanceof org.python.types.Int)) {
+                throw new org.python.exceptions.TypeError("'" + arg2.typeName() + "' object cannot be interpreted as an integer");
+            } else {
+                long m = Math.abs(((org.python.types.Int) arg1).value);
+                long n = Math.abs(((org.python.types.Int) arg2).value);
+                long h = 1;
+                long p = m * n;
+                for (long i = 2; i < p; i++) {
+                    if ((m % i == 0) && (n % i == 0)) {
+                        h = i;
+                    }
+                }
+                return new org.python.types.Int(h);
+            }
+        }
+    }
+
+    @org.python.Method(
+            __doc__ = "",
+            args = {"v"}
+    )
+    public static org.python.Object fabs(org.python.Object v) {
+        return new org.python.types.Float(Math.abs(toFloatValue(v)));
+    }
+
+    @org.python.Method(
+            __doc__ = "",
+            args = {"a1", "a2"}
+    )
+    public static org.python.Object fmod(org.python.Object a1, org.python.Object a2) {
+        double v = toFloatValue(a1);
+        double w = toFloatValue(a2);
+        if (isnan(v) || isnan(w)) {
+            return new org.python.types.Float(Double.NaN);
+        }
+        if (isinf(w)) {
+            return a1;
+        }
+        if (w == 0.0) {
+            throw new org.python.exceptions.ValueError("math domain error");
+        }
+        if (isinf(v) && w == 0.0) {
+            throw new org.python.exceptions.ValueError("math domain error");
+        }
+        return new org.python.types.Float(v % w);
+    }
+
```

Comment:
It would be great if the spacing between all the methods could be made consistent!